app/api/auth/agent-connect/callback/route.ts

```diff
@@ -25,12 +34,49 @@ export const GET = withSession(async function callbackRoute(req) {
     }
   } catch (e: any) {
     logFatalErrorInSentry(new AgentConnectFailedException({ cause: e }));
-    if (e instanceof HttpForbiddenError) {
+    if (e instanceof AgentConnectCouldBeAServicePublicException) {
+      return NextResponse.redirect(
+        getBaseUrl() + '/connexion/habilitation/requise'
+      );
+    } else if (e instanceof HttpForbiddenError) {
       return NextResponse.redirect(
-        getBaseUrl() + '/connexion/echec-authorisation-requise'
+        getBaseUrl() + '/connexion/habilitation/refuse'
       );
     } else {
       return NextResponse.redirect(getBaseUrl() + '/connexion/echec-connexion');
     }
   }
 });
+
+const verifyAgentHabilitation = async (agent: IAgentInfo) => {
+  if (agent.belongToATeam) {
+    return;
+  }
+
+  const { isMCP } = agent;
+
+  // MCP should always return a siret
+  if (isMCP && !agent.siret) {
+    throw new HttpForbiddenError('MCP user must have a siret');
+  }
+
+  const siren = extractSirenFromSiret(agent.siret);
+  const uniteLegale = await getUniteLegaleFromSlug(siren, {
+    page: 0,
+    isBot: false,
+  });
+
+  if (isServicePublic(uniteLegale)) {
+    return;
+  }
+
+  const couldBeServicePublic =
+    uniteLegale.natureJuridique.startsWith('4') ||
+    uniteLegale.natureJuridique.startsWith('8');
+
+  if (couldBeServicePublic) {
+    throw new AgentConnectCouldBeAServicePublicException({});
+  }
```

Comment:
We should comment this as we will wait for the pole juridique to activate this process

```suggestion
  //const couldBeServicePublic =
  //  uniteLegale.natureJuridique.startsWith('4') ||
  //  uniteLegale.natureJuridique.startsWith('8');

  //if (couldBeServicePublic) {
  //  throw new AgentConnectCouldBeAServicePublicException({});
  //}
```